parser_library/src/workspaces/processor_file_impl.cpp

```diff
@@ -39,54 +39,56 @@ bool processor_file_impl::parse(parse_lib_provider& lib_provider,
     std::vector<preprocessor_options> pp,
     virtual_file_monitor* vfm)
 {
-    if (!last_opencode_id_storage_)
-        last_opencode_id_storage_ = std::make_shared<context::id_storage>();
+    if (!m_last_opencode_id_storage)
+        m_last_opencode_id_storage = std::make_shared<context::id_storage>();
 
     const bool collect_hl = should_collect_hl();
     auto fms = std::make_shared<std::vector<fade_message_s>>();
-    auto new_analyzer = std::make_unique<analyzer>(file_->get_text(),
+    auto new_analyzer = std::make_unique<analyzer>(m_file->get_text(),
         analyzer_options {
-            file_->get_location(),
+            m_file->get_location(),
             &lib_provider,
             std::move(asm_opts),
             collect_hl ? collect_highlighting_info::yes : collect_highlighting_info::no,
             file_is_opencode::yes,
-            last_opencode_id_storage_,
+            m_last_opencode_id_storage,
             std::move(pp),
             vfm,
             fms,
         });
-    // If parsed as opencode previously, use id_index from the last parsing
 
-    auto old_dep = dependencies_;
+    auto old_dep = m_dependencies;
 
+    processing::hit_count_analyzer hc_analyzer(new_analyzer->hlasm_ctx());
+    new_analyzer->register_stmt_analyzer(&hc_analyzer);
```

Comment:
this will be eventually a dangling reference, right?
Just add a TODO... I think that eventually we should not keep the analyzers at all, just the various results.

Reply:
That's right 😱  Added todo